src/Migrations/Migration.php

```diff
@@ -1,62 +1,51 @@
- * is identified by the child class name.
+ * Each child class of Migration represents an individual database migration which is identified by the child class
+ * name.
  *
- * Within each migration, the [[up()]] method should be overridden to contain the logic
- * for "upgrading" the database; while the [[down()]] method for the "downgrading"
- * logic. The "yii migrate" command manages all available migrations in an application.
+ * Within each migration, the {@see up()} method should be overridden to contain the logic for "upgrading" the database;
+ * while the {@see down()} method for the "downgrading" logic. The "yii migrate" command manages all available
+ * migrations in an application.
  *
- * If the database supports transactions, you may also override [[safeUp()]] and
- * [[safeDown()]] so that if anything wrong happens during the upgrading or downgrading,
- * the whole migration can be reverted in a whole.
+ * If the database supports transactions, you may also override {@see safeUp()} and {@see safeDown()} so that if
+ * anything wrong happens during the upgrading or downgrading, the whole migration can be reverted in a whole.
  *
- * Note that some DB queries in some DBMS cannot be put into a transaction. For some examples,
- * please refer to [implicit commit](http://dev.mysql.com/doc/refman/5.7/en/implicit-commit.html). If this is the case,
- * you should still implement `up()` and `down()`, instead.
+ * Note that some DB queries in some DBMS cannot be put into a transaction. For some examples, please refer to
+ * [implicit commit](http://dev.mysql.com/doc/refman/5.7/en/implicit-commit.html). If this is the case, you should still
+ * implement `up()` and `down()`, instead.
  *
  * Migration provides a set of convenient methods for manipulating database data and schema.
- * For example, the [[insert()]] method can be used to easily insert a row of data into
- * a database table; the [[createTable()]] method can be used to create a database table.
- * Compared with the same methods in [[Command]], these methods will display extra
- * information showing the method parameters and execution time, which may be useful when
- * applying migrations.
- *
- * For more details and usage information on Migration, see the [guide article on Migration](guide:db-migrations).
  *
- * @author Qiang Xue <qiang.xue@gmail.com>
+ * For example, the {@see insert()} method can be used to easily insert a row of data into a database table; the
+ * {@see createTable()} method can be used to create a database table. Compared with the same methods in {@see Command},
+ * these methods will display extra information showing the method parameters and execution time, which may be useful
+ * when applying migrations.
  *
- * @since 2.0
+ * For more details and usage information on Migration, see the [guide article on Migration](guide:db-migrations).
  */
-class Migration extends Component implements MigrationInterface
+class Migration implements MigrationInterface
```

Comment:
Migrations just be clean, I have not worked on it, we can open a PR apart I think there is a repository to separate it from the db packages and implement it with the Symphony console, since this PR is too large.